juju/juju
# [JUJU-4012] Fix apiserver watchers

The changes to the watcher registry only ensured that the watchers
were workers. Once bootstrapped this caused all sorts of chaos. This
is because no watcher actually aligned with the new type of watcher from
the core package.

The solution was to grab the new FirstResult internal package and shove
it one level higher. Then we can use that from the apiserver package.
From there we then needed to remove all the references from the state
watcher package and use the core ones. Although they're kind of similar
they are in fact not quite the same in terms of behavior. We will have
to fix that once we move everything across.
For now, though, we should just force everything to be a core watcher and
be done with it.

There are virtually zero tests for the watcher package in apiserver and
considering this is the main driver of juju that is actually very
disappointing. We'll have to rectify that in the future.

This will unblock https://github.com/juju/juju/pull/15817 from landing.

## Checklist


## QA steps

Ensure that there are no errors or time outs from the uniter.

```sh
$ juju bootstrap lxd test --build-agent
$ juju add-model default
$ juju deploy ubuntu
$ juju debug-log -m controller
```
